### DISCUSSION TITLE
Add GitHub Copilot provider

### DISCUSSION THREAD

**Discussion by Wolframko** [cline/cline#660] [Feature Requests]
GitHub Copilot subscription is excellent and cheap. It can be integrated into Cline, similar to [codecompanion.nvim](https://github.com/olimorris/codecompanion.nvim/blob/5c5a5c759b8c925e81f8584a0279eefc8a6c6643/lua/codecompanion/adapters/copilot.lua)

**Comment by Jasmin68k** [comment]
Good idea.

**Comment by RodolfoCastanheira** [comment]
Plus, now copilot has Sonnet 3.5

**Comment by Jasmin68k** [comment]
Some more details here: https://github.com/Aider-AI/aider/issues/2227#issuecomment-2453987540.

**Comment by Bigsy** [comment]
I think this is an amazing idea, anyone know why the PR got shutdown for it https://github.com/cline/cline/pull/721 ?

**Comment by shrijayan** [comment]
This will be nice Idea where we can login to github account to use the Github Copilot license and access Cluade3.5

**Comment by dcbartlett** [comment]
This has been implemented.

**Comment by yyovil** [comment]
**uhh uhh!! BAD NEWS!**
<img width="722" height="348" alt="image" src="https://github.com/user-attachments/assets/72ece292-4284-46a9-a04b-d9d40e5b151b" />

MCP is the way now.
